### ISSUE TITLE
CoreOS Layering overwrite modified files in /etc from the layer on rpm-ostree update

### ISSUE THREAD

**Issue by jdoss** [coreos/fedora-coreos-tracker#1589]
### Describe the enhancement

I have been using CoreOS Layering extensively for my workloads and some times in order to fix a bug quickly, I need to modify a file in /etc. I then go back to my automation and update my layer with the fix. Fast forward to updating the same file with new configuration (and the previous fixes) the updated file will not update due to being modified on the server. I am pretty sure this is the intended behavior. 

Is there a way to to tell `rpm-ostree` to overwrite any modified files in /etc that ship in the CoreOS layer or fix the file so it reverts back to the layered version as a work around? Maybe a flag could a flag be added to always use the files in the updated layer? I understand /etc/ is meant to be mutable on FCOS, but I think this use case of always treating files added to the Containerfile after `FROM quay.io/fedora/fedora-coreos:stable` from the layer as the source of truth would be super helpful for operators.

**Comment by dustymabe** [comment]
cc @jmarrero @cgwalters in case you have thoughts on this.

**Comment by jmarrero** [comment]
I think this might be a good use case for: https://github.com/ostreedev/ostree/pull/3062 ?

**Comment by cgwalters** [comment]
Yep exactly, once you go all-in on a custom container image-based flow, having `/etc` be transient just makes sense.

For now you can `ostree admin config-diff` and basically `cp /usr/etc/somefile.conf /etc/somefile.conf`.

But this whole discussion also ties in to https://github.com/coreos/fedora-coreos-tracker/issues/399 which we have new momentum on now (basically we want a `bootc reinstall --keep=/var/lib/containers --keep=/var/log` type flow).

**Comment by jdoss** [comment]
Woah! Heck yeah this is great news.

So one thing comes to mind that I was going to open another issue for is shipping shell users inside containers. I use ignition for sysadmin / emergency users like the `core` user, but adding in other users that could be managed via the container layer would be awesome. I looked into systemd-homed with USER_RECORD for this but it seems there are major SELinux issues for Fedora right now. I am using `sysusers.d` for system users which works great layering. 

Any thoughts on how we can manage local users on a system via the container layer? If we set /etc/ to be transient, that will prevent users from adding and removing users manually, which is what I am currently doing now.
